utils/typename-derive/src/lib.rs

```diff
@@ -38,10 +39,21 @@ pub fn derive(input: TokenStream) -> TokenStream {
     } else {
         let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
         let type_name = ident.to_string();
+        #[allow(unstable_name_collisions)]
+        let gen_params = generics
+            .type_params()
+            .into_iter()
```

Comment:
I'm not sure if it is worth the effort, but probably we can return `Cow::Borrowed` if the list of generic parameters is empty and fall back to `Cow::Owned` otherwise.

Reply:
Not really worth it.